include/grpcpp/impl/codegen/async_stream.h

```diff
@@ -496,6 +496,13 @@ class ClientAsyncReaderWriter final
     assert(size == sizeof(ClientAsyncReaderWriter));
   }
 
+  // This operator should never be called as the memory should be freed as part
+  // of the arena destruction. It only exists to provide a matching operator
+  // delete to the operator new so that some compilers will not complain (see
+  // https://github.com/grpc/grpc/issues/11301) Note at the time of adding this
+  // there are no tests catching the compiler warning.
+  static void operator delete(void*, void*) { assert(0); }
```

Comment:
Should we add the same thing for ClientAsyncReader as well?

Reply:
yeah, perhaps, you know the C++ codebase better than I do. Should I add it?

Reply:
I do not know why vs is complaining while other compilers are not. I assume it will complain about the AsyncReader when it is used. So I would add it to prevent similar issues coming back.